src/shims/unix/linux/foreign_items.rs

```diff
@@ -85,7 +85,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     this.read_scalar(name)?,
                     TASK_COMM_LEN,
                 )?;
-                let res = if res { Scalar::from_u32(0) } else { this.eval_libc("ERANGE") };
+                let res = if res { Scalar::from_u32(0) } else { LibcError("ERANGE") };
```

Comment:
Here, the `if` and `else` have incompatible types. There might be a simple conversion we can wrap the error in.

Reply:
Yeah this doesn't set errno so don't change it.